### PR TITLE
docs: clarify self-signed certificate limitation. 

### PR DESCRIPTION
**Pull request by bdaehlie**
That is a limitation of webpki and should be noted there.

See this [PR for webpki](https://github.com/rustls/webpki/pull/180).

### CODE REVIEW

**Review by cpu — APPROVED** [review]
Thanks!

**Comment by djc** [comment]
I don't think this is a good change.

For most users, webpki is an implementation detail of the rustls default verifier. Even if/when we make the webpki verifier more explicit in the API (for example, by making callers pass a `WebPkiVerifiers::from(roots)` to the config builder API), for a large majority of users rustls is intimately connected with webpki -- that's probably even more true of the relatively unsophisticated users who don't have a good idea about the trade-offs of self-signed certificates.

Instead, I think we should be upfront about this limitation and its nuances.

**Comment by ctz** [comment]
How about this?

* Using CA certificates directly to authenticate a server/client. _CA certificates should
    only issue other certificates, and cannot be used alone for authentication. All authentication can be turned off -- see our [example code](https://github.com/rustls/rustls/blob/a519303f0c4a76cf3ea746e1bae0707b5001b8e0/examples/src/bin/tlsclient-mio.rs#L322).)_

Note I'm carefully avoiding saying "self-signed", which I think can be used to mean three different things:

1. a standalone CA certificate that has CN=localhost (or something) and therefore authenticates itself.
2. a locally-administered PKI (ie, "i signed this myself, it is self-signed")
3. turn-off-all-authentication-I-don't-care (ie, "don't bother me with this certificate bidness")

**Comment by bdaehlie** [comment]
If you're able to take this over that would be great, thanks.

**Comment by cpu** [comment]
@ctz @djc I updated this branch with the suggestions from above. PTAL

**Review by ctz — APPROVED** [review]
(fwiw)

File: README.md [hunk 1]
```diff
@@ -72,7 +72,10 @@ rustls does not and will not support:
 * TLS 1.2 protocol compression.
 * Discrete-log Diffie-Hellman.
 * Automatic protocol version downgrade.
-* Self-signed certificates.
+* Using CA certificates directly to authenticate a server/client (often called "self-signed
+certificates"). _CA certificates should only issue other certificates, and cannot be used
```
Comment (djc) [inline README.md:76]: "cannot be used alone for" seems a little awkward? This is a good direction, though!

**Review by djc — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (cpu) [reply README.md:76]: How about: "CA certificates should only issue other certificates, and are inappropriate for authentication in isolation." ? Less awkward?

Comment (djc) [reply README.md:76]: I don't really grok what you mean by "are inappropriate for authentication in isolation". I guess the real point is that authenticating a server against a certificate signed by its own private key, that is also marked as a CA, would actually be an okay way to authenticate, but we don't want to bear the extra complexity in webpki for enabling path building (safely) scenarios where this is the case?

In which case maybe we should say: "To limit complexity and risk in our path building implementation, our default verifier implementations do not support using an end-entity certificate as a trust anchor."?

Comment (cpu) [reply README.md]: > In which case maybe we should say: "To limit complexity and risk in our path building implementation, our default verifier implementations do not support using an end-entity certificate as a trust anchor."?

Added text closer to this. WDYT?

**Review by djc — APPROVED** [review]

**Pull request merged**